### PR TITLE
add tuya Vibration Sensor ZG-103Z

### PR DESCRIPTION
**Pull request by 13717033460**
add tuya new device ZG-103Z

### CODE REVIEW

**Comment by 13717033460** [comment]
It keeps reporting formatting errors, I'm going crazy

File: src/devices/tuya.ts [hunk 1]
```diff
@@ -11328,6 +11328,35 @@ const definitions: Definition[] = [
...
+    {
+        fingerprint: tuya.fingerprint('TS0601', ['_TZE200_iba1ckek', '_TZE200_hggxgsjj']),
+        model: 'ZG-103Z',
+        vendor: 'Tuya',
+        description: 'Vibration sensor',
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints],
+        configure: tuya.configureMagicPacket,
+        exposes: [
+            e.enum('vibrational_state', ea.STATE, ['none', 'alarm']).withDescription('Sensor vibration alarm'),
```
Comment (Koenkk) [inline src/devices/tuya.ts:11340]:
```suggestion
            e.vibration(),
```

File: src/devices/tuya.ts [hunk 2]
```diff
@@ -11328,6 +11328,35 @@ const definitions: Definition[] = [
...
+            e.enum('tilt_status', ea.STATE, ['none', 'alarm']).withDescription('Sensor tilt alarm'),
+            e.numeric('x', ea.STATE).withValueMin(0).withValueMax(256).withValueStep(1).withDescription('X coordinate'),
+            e.numeric('y', ea.STATE).withValueMin(0).withValueMax(256).withValueStep(1).withDescription('Y coordinate'),
+            e.numeric('z', ea.STATE).withValueMin(0).withValueMax(256).withValueStep(1).withDescription('Z coordinate'),
+            e.battery(),
+            e.enum('sensitivity', ea.STATE_SET, ['low', 'middle', 'high']).withDescription('Vibration detection sensitivity'),
+        ],
+        meta: {
+            tuyaDatapoints: [
+                [1, 'vibrational_state', tuya.valueConverterBasic.lookup({none: tuya.enum(0), alarm: tuya.enum(1)})],
```
Comment (Koenkk) [inline src/devices/tuya.ts:11350]:
```suggestion
                [1, 'vibration', tuya.valueConverter.trueFalseEnum1],
```

File: src/devices/tuya.ts [hunk 3]
```diff
@@ -11328,6 +11328,35 @@ const definitions: Definition[] = [
...
+                [7, 'tilt_status', tuya.valueConverterBasic.lookup({none: tuya.enum(0), alarm: tuya.enum(1)})],
```
Comment (Koenkk) [inline src/devices/tuya.ts:11351]:
```suggestion
                [7, 'tilt', tuya.valueConverter.trueFalseEnum1],
```

File: src/devices/tuya.ts [hunk 4]
```diff
@@ -11328,6 +11328,35 @@ const definitions: Definition[] = [
...
+        fingerprint: tuya.fingerprint('TS0601', ['_TZE200_iba1ckek', '_TZE200_hggxgsjj']),
+        model: 'ZG-103Z',
+        vendor: 'Tuya',
+        description: 'Vibration sensor',
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints],
+        configure: tuya.configureMagicPacket,
+        exposes: [
+            e.enum('vibrational_state', ea.STATE, ['none', 'alarm']).withDescription('Sensor vibration alarm'),
+            e.enum('tilt_status', ea.STATE, ['none', 'alarm']).withDescription('Sensor tilt alarm'),
```
Comment (Koenkk) [inline src/devices/tuya.ts:11341]: Use `e.binary` with `tilt`

Comment (13717033460) [reply src/devices/tuya.ts:11341]: OK,Thanks!!

**Comment by Koenkk** [comment]
Thanks!

Could you also submit a picture for the docs? This can be done by:
1. Create a fork by clicking [here](https://github.com/Koenkk/zigbee2mqtt.io/fork)
2. Go to the `public/images/devices` directory, *Add file* -> *Upload files*
3. Upload the files and press *Commit changes*
4. Press *Contribute* -> *Open pull request* -> update title/description -> *Create pull request*

Make sure that:
- The filename is `ZG-103Z.png`
- The size is 512x512
- The background is transparent (use e.g. [Adobe remove background](https://new.express.adobe.com/tools/remove-background))

**Pull request merged**